### ISSUE TITLE
37.1.0 breaks Vite setup of CKEditor

### ISSUE THREAD

**Issue by Witoso** [ckeditor/ckeditor5#13922]
## 📝 Provide detailed reproduction steps (if any)

1. Project setup with Vite (+ our experimental plugin)
2. Start the Vite dev server. (e.g. `npm run dev`)

### ✔️ Expected result

Server starts.

### ❌ Actual result

Error:

```sh
4:38:23 PM [vite] Internal server error: Failed to parse source for import analysis because the content contains invalid JS syntax. If you are using JSX, make sure to name the file with the .jsx or .tsx extension.
  Plugin: vite:import-analysis
  File: /Users/witold/workspace/cksource/editor/node_modules/.vite/deps/@ckeditor_ckeditor5-table.js?v=b4dc69b9:8456:63
  8454|     * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  8455|     *)
  8456|    (* @if CK_DEBUG //	/* istanbul ignore next -- @preserve */ *)
```

### ❓ Possible solution

Looks like underneath Vite uses some comment that breaks due to our comment.

file: `node_modules/.vite/deps/@ckeditor_ckeditor5-table.js`

**Possibly our ~comment is too close to the license information, and it gets pulled~ the comment has `@` and it's pulled as a license?** 😮‍💨 

```js
/*! Bundled license information:

@ckeditor/ckeditor5-table/src/tableui.js:
  (**
   * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
   * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
   *)

@ckeditor/ckeditor5-table/src/tableselection.js:
  (**
   * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
   * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
   *)
  (* @if CK_DEBUG //	/* istanbul ignore next -- @preserve */ *) <- this is were we break the comment.

@ckeditor/ckeditor5-table/src/tableclipboard.js:
  (**
   * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
   * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
   *)

...

*/ <- end of file

```

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by LukaszGudel** [comment]
The issue comes from the usage of  `@preserve` in istanbul comment. This type of comment is treated as a [legal comment/license comment](https://esbuild.github.io/api/#legal-comments) by esbuild and is bundled to the end of the file, but our custom syntax with `@if CK_DEBUG` results in a broken comment. 

There is only a single place that is using both istanbul ignore comment and CK_DEBUG:
https://github.com/ckeditor/ckeditor5/blob/909e0e354a457d6c4e32aa85fea97b1bd0c3234e/packages/ckeditor5-table/src/tableselection.ts#L92-L96

Other places that also have istanbul comments with @preserve are bundled correctly by Vite/esbuild. 

This issue will affect anyone that:
- is using Vite
- is building the editor from source
- is using 37.1.0 version of editor packages
- and is using ckeditor5-table plugin


## Temporal workaround

Anyone that is using Vite can change the configuration of esbuild in `vite.config.js` and can disable collecting of legal comments during prebuilding of external dependencies:

```js
// vite.config.js

export default defineConfig( {
	optimizeDeps: {
		esbuildOptions: {
			legalComments: 'none'
		}
	},
       // other config options...
} );
```

## Proper fix

Do we need to include any istanbul comments and `CK_DEBUG` comments in files that are released to npm? If we are able to remove these comments before releasing the files, then the issue with Vite would be fixed, and as a bonus point we will not bloat the license output. Currently, if someone would use esbuild to collect all legal comments from all their dependencies then the output would also include our `istanbul ignore` comments.

**Comment by pomek** [comment]
The `/* istanbul ignore next -- @preserve */ ` comment can be removed from the code safely because we do not enable the CK_DEBUG flags in automated tests so there is no chance the coverage will drop.

**Comment by Witoso** [comment]
So to confirm, we need to delete `/* istanbul ignore next -- @preserve */` in every place it's joined with the `CK_DEBUG`.

**Comment by Witoso** [comment]
Scope:
Remove all `/* istanbul ignore next -- @preserve */` that refer to `CK_DEBUG` mode.

Theoretically, we could just remove the one that is on the same line, but it would be better to remove it everywhere as it 1) doesn't do anything, and 2) it may cause issues in the future.

Is there any way we could prevent developers to add it in this place (linter?) to not face it in the future?

**Comment by pomek** [comment]
> Is there any way we could prevent developers to add it in this place (linter?) to not face it in the future?

It is possible to create a rule that bans adding Istanbul comments next to CK_DEBUG flags.

**Comment by YahorDanchanka** [comment]
Have the same issue

**Comment by LukaszGudel** [comment]
Hi @YahorDanchanka,

please take a look at the "Temporal workaround" mentioned in the [comment above](https://github.com/ckeditor/ckeditor5/issues/13922#issuecomment-1516000796) and let us know if this helps.

**Comment by mremiszewski** [comment]
There's a PR with new linter rule. [link](https://github.com/ckeditor/ckeditor5-linters-config/pull/15)

**Comment by lmcsu** [comment]
Even VSCode shows that it's an incorrect syntax

![image](https://github.com/ckeditor/ckeditor5/assets/34189985/40f867b2-65c6-4706-83d0-b2e94894293c)

**Comment by Witoso** [comment]
As mentioned above, one of our comments breaks the legal comments. This will be fixed in v38.0.0.
